src/main/java/net/snowflake/client/jdbc/SnowflakeConnectString.java

```diff
@@ -130,18 +130,17 @@ public static SnowflakeConnectString parse(String url, Properties info) {
         // if it's a global url
         parameters.put("ACCOUNT", account);
       }
-
-      if (account.contains("_")
-          && parameters.containsKey(
-              SFSessionProperty.ALLOW_UNDERSCORES_IN_HOST.getPropertyKey().toUpperCase())
-          && "false"
-              .equalsIgnoreCase(
-                  (String)
-                      parameters.get(
-                          SFSessionProperty.ALLOW_UNDERSCORES_IN_HOST
-                              .getPropertyKey()
-                              .toUpperCase()))
-          && host.startsWith(account)) {
+      boolean allowUnderscoresInHost = false;
```

Comment:
@sfc-gh-ext-simba-lb can you please add some comments here explaining what is going on? Thanks.